internal/transport/transport_test.go

```diff
@@ -476,10 +486,42 @@ func setUpWithNoPingServer(t *testing.T, copts ConnectOptions, connCh chan net.C
 		}
 		framer := http2.NewFramer(conn, conn)
 		if err := framer.WriteSettings(); err != nil {
-			t.Errorf("Error at server-side while writing settings: %v", err)
+			t.Errorf("Error at server-side while writing settings: %v", framer.ErrorDetail())
 			close(connCh)
 			return
 		}
+		if respModeCh != nil {
+			if err := framer.WriteSettingsAck(); err != nil {
+				t.Errorf("Error while writing settings: %v", err)
+				return
+			}
+			go func() {
+				respondToPing := true
+				var lock sync.Mutex
```

Comment:
This lock doesn't seem to do anything at all

Reply:
This lock protects the https://github.com/grpc/grpc-go/blob/fc7470388ccd16b300866941e8532797c329dd37/internal/transport/transport_test.go#L518

Reply:
Good catch! I'll fix that. Thanks